<center>
<b><u>Predictive Models</u></b>
<br>
Final Project - Data Cleaning
<br>
Michael Rawat
<br>
07/05/2018
</center>

<br>

<center>
Imbalanced Data Sets
</center>

### Contents

1) Importing Packages & General

2) Churn Data Set
    - Numeric and Non Numeric Variables
    - Missing Data
    - Relocating and Converting Variables
    - Conducting Min-Max Standardization
    - Target Variable and Predictor Sets
    - Exporting Cleaned Data
    
3) Autism Data Set
    - Numeric and Non Numeric Variables
    - Missing Data
    - Relocating and Converting Variables
    - Conducting Min-Max Standardization
    - Target Variable and Predictor Sets
    - Exporting Cleaned Data

4) Heart Attacks Data Set
    - Numeric and Non Numeric Variables
    - Missing Data
    - Relocating and Converting Variables
    - Conducting Min-Max Standardization
    - Target Variable and Predictor Sets
    - Exporting Cleaned Data
    
5) Post Operation Data Set
    - Numeric and Non Numeric Variables
    - Missing Data
    - Relocating and Converting Variables
    - Conducting Min-Max Standardization
    - Target Variable and Predictor Sets
    - Exporting Cleaned Data

6) Learning More
    - Numeric and Non Numeric Variables
    - Missing Data
    - Relocating and Converting Variables
    - Conducting Min-Max Standardization
    - Target Variable and Predictor Sets
    - Exporting Cleaned Data

### Importing Packages & General

In [30]:
# Importing package necessary for Data frames
import pandas as pd

# Importing package necessary for calculations
import numpy as np

# Importing package to convert a text variable into a numeric variable
from sklearn import preprocessing

# Defining pathname of working directory
pathname = "C:\\Users\\Michael's Desktop\\OneDrive - Rockhurst University\\Graduate School\\2018 - Summer\\BIA 6303 - Predictive Models\\Homework\\Final Project\\"

# Setting the display width to view the entire text
pd.set_option('display.max_colwidth', 150)

### Churn Data Set

In [31]:
# Importing Churn data set
Churn = pd.read_csv(pathname + 'Datasets\\Churn.csv', encoding = 'utf-8')

# Checking data set was imported by printing first and last five records
print('\r')
print("Below shows the first five records in the data set:")
display(Churn.head())

# Checking the size of the data set
print('\r')
print("The data set has x records across y variables (x,y):")
print(Churn.shape)

######################### Numeric and Non-Numeric Variables #########################

# Describing all numeric variables
print('\r')
print("Below is information about every numeric variable present in the data set:")
display(Churn.describe(include=[np.number]))
print('\r')
print("Notice that "+str(Churn.describe(include=[np.number]).shape[1])+" columns of data contain numeric data.")

# Describing all non-numeric variables
if (Churn.describe(include=[np.number]).shape[1] <  Churn.shape[1]):
    print('\r')
    print("Below is information about every non-numeric variable present in the data set:")
    display(Churn.describe(exclude=[np.number]))
    print('\r')
    print("Notice that "+str(Churn.describe(exclude=[np.number]).shape[1])+" columns of data contain non-numeric data.")
else:
    print('\r')
    print("There are no non-numeric variables within the data set.")
    
######################### Missing Data #########################

# Creating a count of missing values in each column
missing_values = Churn.isnull().sum()

# Displaying the number of missing values in each column that contains a missing value
print('\r')
print("The variables below contain missing values:")
display(missing_values[missing_values>0])

######################### Re-locating and Converting Variables #########################

# Designating Target Variable
Target_Variable = 'Churn Flag'
Target_Series = Churn[Target_Variable]

# Removing target variable from current location and inserting in collum 0
del Churn[Target_Variable]
Churn.insert(0, Target_Variable, Target_Series)
        
# Showing the data frame after target variable has been moved to the front
print('\r')
print("Below shows the first five records in the data set after the target variable has been moved and converted:")
display(Churn.head())

# Converting predictor into numeric variable for learning algorithms later
numeric_encoder = preprocessing.LabelEncoder()
Churn['AON'] = numeric_encoder.fit_transform(Churn['AON'])

######################### Conducting Min-Max Standardization #########################

# Creating a list of numeric variable names
List_of_Numeric_Variables = Churn.describe(include=[np.number]).columns.tolist()

# Creating a for loop to min-max standardize every numeric variable
for variable in List_of_Numeric_Variables:
    Churn[variable] = (Churn[variable]-Churn[variable].min())/(Churn[variable].max()-Churn[variable].min())

# Displaying updated numeric variable statistics to check min max standardization was successful
print('\r')
print("Below is an updated table of numeric variable statistics to demonstrate Min-Max Standardization has been successful:")
display(Churn.describe(include=[np.number]))

######################### Target Variable and Predictor Sets #########################

# Defining Target Variable Column
target_variable_column = Churn[Target_Variable]

# Checking the size of the target variable data set
print('\r')
print("The target variable data set (x,y) contains x records across y features:")
print(target_variable_column.shape)
print('\r')
print("Below is a list of all the variables present in the target variable data set:")
print('\r')
print(target_variable_column.name)

# Creating a data set contains all the predictor variables
predictor_variables_columns = Churn.drop([Target_Variable], axis=1) 

# Checking the size of the target variable data set
print('\r')
print("The predictor variables data set (x,y) contains x records across y features:")
print(predictor_variables_columns.shape)
print('\r')
print("Below is a list of all the variables present in the predictor variable data set:")
print('\r')
print(predictor_variables_columns.columns)

# Storing target variable and predictor variables to arrays for PCA
target_variable_array = target_variable_column.values
predictor_variables_array = predictor_variables_columns.values

######################### Exporting Cleaned Data Set for Analysis #########################

# Exporting cleaned data frame as csv
Churn.to_csv('Churn_Cleaned.csv', index=False)


Below shows the first five records in the data set:


MSISDN           AON  Churn Flag  Avg_Inactive_Days in last 1 month  \
0  7610120456   6-12 months           1                               4.18   
1  7611120467  12-18 months           0                              16.00   
2  9711120467    >18 months           0                              18.00   
3  9830022336    3-6 months           0                              18.00   
4  9713246555    >18 months           0                               6.00   

   Current_Balance in INR  Data Days Latest week  Calling Days Latestweek  \
0                      20                      1                        1   
1                      18                      2                        3   
2                       8                      4                        2   
3                       8                      3                        2   
4                      44                      6                        3   

   Recharge Amount Latest Week  Avg_OG_Local_Offnet_Cnt  \
0                           44                       13   
1                           18                        7   
2                           98                       12   
3                           86                        3   
4                           36                       11   

   Avg_OG_Local_Offnet_Cnt.1  Roaming User  \
0                          5             0   
1                         14             1   
2                         12             0   
3                         12             0   
4                          4             0   

   Avergae Complaints Count Last 4 weeks  
0                                      3  
1                                      1  
2                                      0  
3                                      2  
4                                      2


The data set has x records across y variables (x,y):
(196, 12)

Below is information about every numeric variable present in the data set:


MSISDN  Churn Flag  Avg_Inactive_Days in last 1 month  \
count  1.960000e+02  196.000000                         196.000000   
mean   8.876168e+09    0.015306                          14.873367   
std    6.492464e+08    0.123082                           6.535869   
min    7.610120e+09    0.000000                           4.000000   
25%    8.366588e+09    0.000000                           8.750000   
50%    8.861838e+09    0.000000                          16.000000   
75%    9.417552e+09    0.000000                          21.000000   
max    9.993631e+09    1.000000                          25.000000   

       Current_Balance in INR  Data Days Latest week  Calling Days Latestweek  \
count              196.000000             196.000000               196.000000   
mean                32.739796               4.102041                 4.448980   
std                 17.305361               1.902710                 1.686279   
min                  1.000000               1.000000                 1.000000   
25%                 18.000000               2.750000                 3.000000   
50%                 34.500000               4.000000                 4.500000   
75%                 47.000000               6.000000                 6.000000   
max                 94.000000               7.000000                 7.000000   

       Recharge Amount Latest Week  Avg_OG_Local_Offnet_Cnt  \
count                   196.000000               196.000000   
mean                     55.867347                 9.698980   
std                      26.487206                 3.911351   
min                      11.000000                 3.000000   
25%                      33.750000                 6.000000   
50%                      54.500000                10.000000   
75%                      79.250000                13.000000   
max                     100.000000                16.000000   

       Avg_OG_Local_Offnet_Cnt.1  Roaming User  \
count                 196.000000    196.000000   
mean                   10.954082      0.153061   
std                     5.102331      0.360969   
min                     3.000000      0.000000   
25%                     6.750000      0.000000   
50%                    11.000000      0.000000   
75%                    15.250000      0.000000   
max                    20.000000      1.000000   

       Avergae Complaints Count Last 4 weeks  
count                             196.000000  
mean                                1.469388  
std                                 1.134124  
min                                 0.000000  
25%                                 0.000000  
50%                                 1.000000  
75%                                 3.000000  
max                                 3.000000


Notice that 11 columns of data contain numeric data.

Below is information about every non-numeric variable present in the data set:


AON
count          196
unique           4
top     >18 months
freq            78


Notice that 1 columns of data contain non-numeric data.

The variables below contain missing values:


Series([], dtype: int64)


Below shows the first five records in the data set after the target variable has been moved and converted:


Churn Flag      MSISDN           AON  Avg_Inactive_Days in last 1 month  \
0           1  7610120456   6-12 months                               4.18   
1           0  7611120467  12-18 months                              16.00   
2           0  9711120467    >18 months                              18.00   
3           0  9830022336    3-6 months                              18.00   
4           0  9713246555    >18 months                               6.00   

   Current_Balance in INR  Data Days Latest week  Calling Days Latestweek  \
0                      20                      1                        1   
1                      18                      2                        3   
2                       8                      4                        2   
3                       8                      3                        2   
4                      44                      6                        3   

   Recharge Amount Latest Week  Avg_OG_Local_Offnet_Cnt  \
0                           44                       13   
1                           18                        7   
2                           98                       12   
3                           86                        3   
4                           36                       11   

   Avg_OG_Local_Offnet_Cnt.1  Roaming User  \
0                          5             0   
1                         14             1   
2                         12             0   
3                         12             0   
4                          4             0   

   Avergae Complaints Count Last 4 weeks  
0                                      3  
1                                      1  
2                                      0  
3                                      2  
4                                      2


Below is an updated table of numeric variable statistics to demonstrate Min-Max Standardization has been successful:


Churn Flag      MSISDN         AON  Avg_Inactive_Days in last 1 month  \
count  196.000000  196.000000  196.000000                         196.000000   
mean     0.015306    0.531169    0.600340                           0.517779   
std      0.123082    0.272391    0.388759                           0.311232   
min      0.000000    0.000000    0.000000                           0.000000   
25%      0.000000    0.317375    0.333333                           0.226190   
50%      0.000000    0.525157    0.666667                           0.571429   
75%      0.000000    0.758306    1.000000                           0.809524   
max      1.000000    1.000000    1.000000                           1.000000   

       Current_Balance in INR  Data Days Latest week  Calling Days Latestweek  \
count              196.000000             196.000000               196.000000   
mean                 0.341288               0.517007                 0.574830   
std                  0.186079               0.317118                 0.281047   
min                  0.000000               0.000000                 0.000000   
25%                  0.182796               0.291667                 0.333333   
50%                  0.360215               0.500000                 0.583333   
75%                  0.494624               0.833333                 0.833333   
max                  1.000000               1.000000                 1.000000   

       Recharge Amount Latest Week  Avg_OG_Local_Offnet_Cnt  \
count                   196.000000               196.000000   
mean                      0.504127                 0.515306   
std                       0.297609                 0.300873   
min                       0.000000                 0.000000   
25%                       0.255618                 0.230769   
50%                       0.488764                 0.538462   
75%                       0.766854                 0.769231   
max                       1.000000                 1.000000   

       Avg_OG_Local_Offnet_Cnt.1  Roaming User  \
count                 196.000000    196.000000   
mean                    0.467887      0.153061   
std                     0.300137      0.360969   
min                     0.000000      0.000000   
25%                     0.220588      0.000000   
50%                     0.470588      0.000000   
75%                     0.720588      0.000000   
max                     1.000000      1.000000   

       Avergae Complaints Count Last 4 weeks  
count                             196.000000  
mean                                0.489796  
std                                 0.378041  
min                                 0.000000  
25%                                 0.000000  
50%                                 0.333333  
75%                                 1.000000  
max                                 1.000000


The target variable data set (x,y) contains x records across y features:
(196,)

Below is a list of all the variables present in the target variable data set:

Churn Flag

The predictor variables data set (x,y) contains x records across y features:
(196, 11)

Below is a list of all the variables present in the predictor variable data set:

Index(['MSISDN', 'AON', 'Avg_Inactive_Days in last 1 month',
       'Current_Balance in INR', 'Data Days Latest week',
       'Calling Days Latestweek', 'Recharge Amount Latest Week',
       'Avg_OG_Local_Offnet_Cnt', 'Avg_OG_Local_Offnet_Cnt.1', 'Roaming User',
       'Avergae Complaints Count Last 4 weeks'],
      dtype='object')


### Churn Data Set Pre-Processing Summary

> <b>Initial</b>
- 196 records by 12 variables
- 11 Numeric Variables and 1 Non-Numeric Variable
- Variety of Ranges

> <b>Cleaned</b>
- 196 records by 12 variables
- 12 Numeric Variables
- Min-Max Standardized

### Autism Data Set

In [32]:
# Importing Autism data set
Autism = pd.read_csv(pathname + 'Datasets\\Autism.csv', encoding = 'utf-8')

# Checking data set was imported by printing first and last five records
print('\r')
print("Below shows the first five records in the data set:")
display(Autism.head())

# Checking the size of the data set
print('\r')
print("The data set has x records across y variables (x,y):")
print(Autism.shape)

######################### Numeric and Non-Numeric Variables #########################

# Describing all numeric variables
print('\r')
print("Below is information about every numeric variable present in the data set:")
display(Autism.describe(include=[np.number]))
print('\r')
print("Notice that "+str(Autism.describe(include=[np.number]).shape[1])+" columns of data contain numeric data.")

# Describing all non-numeric variables
if (Autism.describe(include=[np.number]).shape[1] <  Autism.shape[1]):
    print('\r')
    print("Below is information about every non-numeric variable present in the data set:")
    display(Autism.describe(exclude=[np.number]))
    print('\r')
    print("Notice that "+str(Autism.describe(exclude=[np.number]).shape[1])+" columns of data contain non-numeric data.")
else:
    print('\r')
    print("There are no non-numeric variables within the data set.")
    
######################### Missing Data #########################

# Creating a count of missing values in each column
missing_values = Autism.isnull().sum()

# Displaying the number of missing values in each column that contains a missing value
print('\r')
print("The variables below contain missing values:")
display(missing_values[missing_values>0])

######################### Re-locating and Converting Variables #########################

# Designating Target Variable
Target_Variable = 'Class/ASD'
Target_Series = Autism[Target_Variable]

# Removing target variable from current location and inserting in collum 0
del Autism[Target_Variable]
Autism.insert(0, Target_Variable, Target_Series)
        
# Showing the data frame after target variable has been moved to the front
print('\r')
print("Below shows the first five records in the data set after the target variable has been moved and converted:")
display(Autism.head())

# Converting target and predictor into numeric variable for learning algorithms later
numeric_encoder = preprocessing.LabelEncoder()
Autism['Class/ASD'] = numeric_encoder.fit_transform(Autism['Class/ASD'])
Autism['age'] = numeric_encoder.fit_transform(Autism['age'])
Autism['gender'] = numeric_encoder.fit_transform(Autism['gender'])
Autism['ethnicity'] = numeric_encoder.fit_transform(Autism['ethnicity'])
Autism['jundice'] = numeric_encoder.fit_transform(Autism['jundice'])
Autism['austim'] = numeric_encoder.fit_transform(Autism['austim'])
Autism['contry_of_res'] = numeric_encoder.fit_transform(Autism['contry_of_res'])
Autism['used_app_before'] = numeric_encoder.fit_transform(Autism['used_app_before'])
Autism['age_desc'] = numeric_encoder.fit_transform(Autism['age_desc'])
Autism['relation'] = numeric_encoder.fit_transform(Autism['relation'])

######################### Conducting Min-Max Standardization #########################

# Creating a list of numeric variable names
List_of_Numeric_Variables = Autism.describe(include=[np.number]).columns.tolist()

# Creating a for loop to min-max standardize every numeric variable
for variable in List_of_Numeric_Variables:
    Autism[variable] = (Autism[variable]-Autism[variable].min())/(Autism[variable].max()-Autism[variable].min())

# Displaying updated numeric variable statistics to check min max standardization was successful
print('\r')
print("Below is an updated table of numeric variable statistics to demonstrate Min-Max Standardization has been successful:")
display(Autism.describe(include=[np.number]))

######################### Target Variable and Predictor Sets #########################

# Defining Target Variable Column
target_variable_column = Autism[Target_Variable]

# Checking the size of the target variable data set
print('\r')
print("The target variable data set (x,y) contains x records across y features:")
print(target_variable_column.shape)
print('\r')
print("Below is a list of all the variables present in the target variable data set:")
print('\r')
print(target_variable_column.name)

# Creating a data set contains all the predictor variables
predictor_variables_columns = Autism.drop([Target_Variable], axis=1) 

# Checking the size of the target variable data set
print('\r')
print("The predictor variables data set (x,y) contains x records across y features:")
print(predictor_variables_columns.shape)
print('\r')
print("Below is a list of all the variables present in the predictor variable data set:")
print('\r')
print(predictor_variables_columns.columns)

# Storing target variable and predictor variables to arrays for PCA
target_variable_array = target_variable_column.values
predictor_variables_array = predictor_variables_columns.values

######################### Exporting Cleaned Data Set for Analysis #########################

# Exporting cleaned data frame as csv
Autism.to_csv('Autism_Cleaned.csv', index=False)


Below shows the first five records in the data set:


A1_Score  A2_Score  A3_Score  A4_Score  A5_Score  A6_Score  A7_Score  \
0         1         1         1         1         0         0         1   
1         1         1         0         1         0         0         0   
2         1         1         0         1         1         0         1   
3         1         1         0         1         0         0         1   
4         1         0         0         0         0         0         0   

   A8_Score  A9_Score  A10_Score    ...    gender       ethnicity jundice  \
0         1         0          0    ...         f  White-European      no   
1         1         0          1    ...         m          Latino      no   
2         1         1          1    ...         m          Latino     yes   
3         1         0          1    ...         f  White-European      no   
4         1         0          0    ...         f               ?      no   

  austim  contry_of_res used_app_before result     age_desc relation Class/ASD  
0     no  United States              no      6  18 and more     Self        NO  
1    yes         Brazil              no      5  18 and more     Self        NO  
2    yes          Spain              no      8  18 and more   Parent       YES  
3    yes  United States              no      6  18 and more     Self        NO  
4     no          Egypt              no      2  18 and more        ?        NO  

[5 rows x 21 columns]


The data set has x records across y variables (x,y):
(704, 21)

Below is information about every numeric variable present in the data set:


A1_Score    A2_Score    A3_Score    A4_Score    A5_Score    A6_Score  \
count  704.000000  704.000000  704.000000  704.000000  704.000000  704.000000   
mean     0.721591    0.453125    0.457386    0.495739    0.498580    0.284091   
std      0.448535    0.498152    0.498535    0.500337    0.500353    0.451301   
min      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
50%      1.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
75%      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   
max      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   

         A7_Score    A8_Score    A9_Score   A10_Score      result  
count  704.000000  704.000000  704.000000  704.000000  704.000000  
mean     0.417614    0.649148    0.323864    0.573864    4.875000  
std      0.493516    0.477576    0.468281    0.494866    2.501493  
min      0.000000    0.000000    0.000000    0.000000    0.000000  
25%      0.000000    0.000000    0.000000    0.000000    3.000000  
50%      0.000000    1.000000    0.000000    1.000000    4.000000  
75%      1.000000    1.000000    1.000000    1.000000    7.000000  
max      1.000000    1.000000    1.000000    1.000000   10.000000


Notice that 11 columns of data contain numeric data.

Below is information about every non-numeric variable present in the data set:


age gender       ethnicity jundice austim  contry_of_res  \
count   704    704             704     704    704            704   
unique   47      2              12       2      2             67   
top      21      m  White-European      no     no  United States   
freq     49    367             233     635    613            113   

       used_app_before     age_desc relation Class/ASD  
count              704          704      704       704  
unique               2            1        6         2  
top                 no  18 and more     Self        NO  
freq               692          704      522       515


Notice that 10 columns of data contain non-numeric data.

The variables below contain missing values:


Series([], dtype: int64)


Below shows the first five records in the data set after the target variable has been moved and converted:


Class/ASD  A1_Score  A2_Score  A3_Score  A4_Score  A5_Score  A6_Score  \
0        NO         1         1         1         1         0         0   
1        NO         1         1         0         1         0         0   
2       YES         1         1         0         1         1         0   
3        NO         1         1         0         1         0         0   
4        NO         1         0         0         0         0         0   

   A7_Score  A8_Score  A9_Score   ...     age gender       ethnicity jundice  \
0         1         1         0   ...      26      f  White-European      no   
1         0         1         0   ...      24      m          Latino      no   
2         1         1         1   ...      27      m          Latino     yes   
3         1         1         0   ...      35      f  White-European      no   
4         0         1         0   ...      40      f               ?      no   

  austim  contry_of_res used_app_before result     age_desc relation  
0     no  United States              no      6  18 and more     Self  
1    yes         Brazil              no      5  18 and more     Self  
2    yes          Spain              no      8  18 and more   Parent  
3    yes  United States              no      6  18 and more     Self  
4     no          Egypt              no      2  18 and more        ?  

[5 rows x 21 columns]


Below is an updated table of numeric variable statistics to demonstrate Min-Max Standardization has been successful:


Class/ASD    A1_Score    A2_Score    A3_Score    A4_Score    A5_Score  \
count  704.000000  704.000000  704.000000  704.000000  704.000000  704.000000   
mean     0.268466    0.721591    0.453125    0.457386    0.495739    0.498580   
std      0.443476    0.448535    0.498152    0.498535    0.500337    0.500353   
min      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
50%      0.000000    1.000000    0.000000    0.000000    0.000000    0.000000   
75%      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   
max      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   

         A6_Score    A7_Score    A8_Score    A9_Score     ...             age  \
count  704.000000  704.000000  704.000000  704.000000     ...      704.000000   
mean     0.284091    0.417614    0.649148    0.323864     ...        0.270936   
std      0.451301    0.493516    0.477576    0.468281     ...        0.220052   
min      0.000000    0.000000    0.000000    0.000000     ...        0.000000   
25%      0.000000    0.000000    0.000000    0.000000     ...        0.086957   
50%      0.000000    0.000000    1.000000    0.000000     ...        0.217391   
75%      1.000000    1.000000    1.000000    1.000000     ...        0.391304   
max      1.000000    1.000000    1.000000    1.000000     ...        1.000000   

           gender   ethnicity     jundice      austim  contry_of_res  \
count  704.000000  704.000000  704.000000  704.000000     704.000000   
mean     0.521307    0.482309    0.098011    0.129261       0.667420   
std      0.499901    0.361160    0.297541    0.335728       0.292502   
min      0.000000    0.000000    0.000000    0.000000       0.000000   
25%      0.000000    0.090909    0.000000    0.000000       0.439394   
50%      1.000000    0.454545    0.000000    0.000000       0.651515   
75%      1.000000    0.909091    0.000000    0.000000       0.954545   
max      1.000000    1.000000    1.000000    1.000000       1.000000   

       used_app_before      result  age_desc    relation  
count       704.000000  704.000000       0.0  704.000000  
mean          0.017045    0.487500       NaN    0.819886  
std           0.129533    0.250149       NaN    0.349032  
min           0.000000    0.000000       NaN    0.000000  
25%           0.000000    0.300000       NaN    0.800000  
50%           0.000000    0.400000       NaN    1.000000  
75%           0.000000    0.700000       NaN    1.000000  
max           1.000000    1.000000       NaN    1.000000  

[8 rows x 21 columns]


The target variable data set (x,y) contains x records across y features:
(704,)

Below is a list of all the variables present in the target variable data set:

Class/ASD

The predictor variables data set (x,y) contains x records across y features:
(704, 20)

Below is a list of all the variables present in the predictor variable data set:

Index(['A1_Score', 'A2_Score', 'A3_Score', 'A4_Score', 'A5_Score', 'A6_Score',
       'A7_Score', 'A8_Score', 'A9_Score', 'A10_Score', 'age', 'gender',
       'ethnicity', 'jundice', 'austim', 'contry_of_res', 'used_app_before',
       'result', 'age_desc', 'relation'],
      dtype='object')


### Autism Data Set Pre-Processing Summary

> <b>Initial</b>
- 704 records by 21 variables
- 11 Numeric Variables and 10 Non-Numeric Variables
- Variety of Ranges

> <b>Cleaned</b>
- 704 records by 21 variables
- 21 Numeric Variables
- Min-Max Standardized

### Heart Attacks Data Set

In [33]:
# Importing Heart Attacks data set
Heart_Attacks = pd.read_csv(pathname + 'Datasets\\Heart Attacks.csv', encoding = 'utf-8', na_values=["?"])

# Checking data set was imported by printing first and last five records
print('\r')
print("Below shows the first five records in the data set:")
display(Heart_Attacks.head())

# Checking the size of the data set
print('\r')
print("The data set has x records across y variables (x,y):")
print(Heart_Attacks.shape)

######################### Numeric and Non-Numeric Variables #########################

# Describing all numeric variables
print('\r')
print("Below is information about every numeric variable present in the data set:")
display(Heart_Attacks.describe(include=[np.number]))
print('\r')
print("Notice that "+str(Heart_Attacks.describe(include=[np.number]).shape[1])+" columns of data contain numeric data.")

# Describing all non-numeric variables
if (Heart_Attacks.describe(include=[np.number]).shape[1] <  Heart_Attacks.shape[1]):
    print('\r')
    print("Below is information about every non-numeric variable present in the data set:")
    display(Heart_Attacks.describe(exclude=[np.number]))
    print('\r')
    print("Notice that "+str(Heart_Attacks.describe(exclude=[np.number]).shape[1])+" columns of data contain non-numeric data.")
else:
    print('\r')
    print("There are no non-numeric variables within the data set.")
    
######################### Missing Data #########################

# Creating a count of missing values in each column
missing_values = Heart_Attacks.isnull().sum()

# Displaying the number of missing values in each column that contains a missing value
print('\r')
print("The variables below contain missing values:")
display(missing_values[missing_values>0])

# Removing variables with significant missing data
del Heart_Attacks['slope']
del Heart_Attacks['ca']
del Heart_Attacks['thal']

# Dropping all rows with a missing value
Heart_Attacks = Heart_Attacks.dropna()

######################### Re-locating and Converting Variables #########################

# Designating Target Variable
Target_Variable = 'num'
Target_Series = Heart_Attacks[Target_Variable]

# Removing target variable from current location and inserting in collum 0
del Heart_Attacks[Target_Variable]
Heart_Attacks.insert(0, Target_Variable, Target_Series)
        
# Showing the data frame after target variable has been moved to the front
print('\r')
print("Below shows the first five records in the data set after the target variable has been moved and converted:")
display(Heart_Attacks.head())

######################### Conducting Min-Max Standardization #########################

# Creating a list of numeric variable names
List_of_Numeric_Variables = Heart_Attacks.describe(include=[np.number]).columns.tolist()

# Creating a for loop to min-max standardize every numeric variable
for variable in List_of_Numeric_Variables:
    Heart_Attacks[variable] = (Heart_Attacks[variable]-Heart_Attacks[variable].min())/(Heart_Attacks[variable].max()-Heart_Attacks[variable].min())

# Displaying updated numeric variable statistics to check min max standardization was successful
print('\r')
print("Below is an updated table of numeric variable statistics to demonstrate Min-Max Standardization has been successful:")
display(Heart_Attacks.describe(include=[np.number]))

######################### Target Variable and Predictor Sets #########################

# Defining Target Variable Column
target_variable_column = Heart_Attacks[Target_Variable]

# Checking the size of the target variable data set
print('\r')
print("The target variable data set (x,y) contains x records across y features:")
print(target_variable_column.shape)
print('\r')
print("Below is a list of all the variables present in the target variable data set:")
print('\r')
print(target_variable_column.name)

# Creating a data set contains all the predictor variables
predictor_variables_columns = Heart_Attacks.drop([Target_Variable], axis=1) 

# Checking the size of the target variable data set
print('\r')
print("The predictor variables data set (x,y) contains x records across y features:")
print(predictor_variables_columns.shape)
print('\r')
print("Below is a list of all the variables present in the predictor variable data set:")
print('\r')
print(predictor_variables_columns.columns)

# Storing target variable and predictor variables to arrays for PCA
target_variable_array = target_variable_column.values
predictor_variables_array = predictor_variables_columns.values

######################### Exporting Cleaned Data Set for Analysis #########################

# Exporting cleaned data frame as csv
Heart_Attacks.to_csv('Heart_Attacks_Cleaned.csv', index=False)


Below shows the first five records in the data set:


age  sex  cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak  \
0   28    1   2     130.0  132.0  0.0      2.0    185.0    0.0      0.0   
1   29    1   2     120.0  243.0  0.0      0.0    160.0    0.0      0.0   
2   29    1   2     140.0    NaN  0.0      0.0    170.0    0.0      0.0   
3   30    0   1     170.0  237.0  0.0      1.0    170.0    0.0      0.0   
4   31    0   2     100.0  219.0  0.0      1.0    150.0    0.0      0.0   

   slope  ca  thal  num  
0    NaN NaN   NaN    0  
1    NaN NaN   NaN    0  
2    NaN NaN   NaN    0  
3    NaN NaN   6.0    0  
4    NaN NaN   NaN    0


The data set has x records across y variables (x,y):
(294, 14)

Below is information about every numeric variable present in the data set:


age         sex          cp    trestbps        chol         fbs  \
count  294.000000  294.000000  294.000000  293.000000  271.000000  286.000000   
mean    47.826531    0.724490    2.982993  132.583618  250.848708    0.069930   
std      7.811812    0.447533    0.965117   17.626568   67.657711    0.255476   
min     28.000000    0.000000    1.000000   92.000000   85.000000    0.000000   
25%     42.000000    0.000000    2.000000  120.000000  209.000000    0.000000   
50%     49.000000    1.000000    3.000000  130.000000  243.000000    0.000000   
75%     54.000000    1.000000    4.000000  140.000000  282.500000    0.000000   
max     66.000000    1.000000    4.000000  200.000000  603.000000    1.000000   

          restecg     thalach       exang     oldpeak       slope   ca  \
count  293.000000  293.000000  293.000000  294.000000  104.000000  3.0   
mean     0.218430  139.129693    0.303754    0.586054    1.894231  0.0   
std      0.460868   23.589749    0.460665    0.908648    0.338995  0.0   
min      0.000000   82.000000    0.000000    0.000000    1.000000  0.0   
25%      0.000000  122.000000    0.000000    0.000000    2.000000  0.0   
50%      0.000000  140.000000    0.000000    0.000000    2.000000  0.0   
75%      0.000000  155.000000    1.000000    1.000000    2.000000  0.0   
max      2.000000  190.000000    1.000000    5.000000    3.000000  0.0   

            thal         num  
count  28.000000  294.000000  
mean    5.642857    0.360544  
std     1.615074    0.480977  
min     3.000000    0.000000  
25%     5.250000    0.000000  
50%     6.000000    0.000000  
75%     7.000000    1.000000  
max     7.000000    1.000000


Notice that 14 columns of data contain numeric data.

There are no non-numeric variables within the data set.

The variables below contain missing values:


trestbps      1
chol         23
fbs           8
restecg       1
thalach       1
exang         1
slope       190
ca          291
thal        266
dtype: int64


Below shows the first five records in the data set after the target variable has been moved and converted:


num  age  sex  cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak
0    0   28    1   2     130.0  132.0  0.0      2.0    185.0    0.0      0.0
1    0   29    1   2     120.0  243.0  0.0      0.0    160.0    0.0      0.0
3    0   30    0   1     170.0  237.0  0.0      1.0    170.0    0.0      0.0
4    0   31    0   2     100.0  219.0  0.0      1.0    150.0    0.0      0.0
5    0   32    0   2     105.0  198.0  0.0      0.0    165.0    0.0      0.0


Below is an updated table of numeric variable statistics to demonstrate Min-Max Standardization has been successful:


num         age         sex          cp    trestbps        chol  \
count  261.000000  261.000000  261.000000  261.000000  261.000000  261.000000   
mean     0.375479    0.534327    0.735632    0.667944    0.376295    0.316262   
std      0.485177    0.211953    0.441843    0.323573    0.163980    0.126772   
min      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000    0.378378    0.000000    0.333333    0.259259    0.237452   
50%      0.000000    0.567568    1.000000    0.666667    0.351852    0.303089   
75%      1.000000    0.702703    1.000000    1.000000    0.444444    0.376448   
max      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   

              fbs     restecg     thalach       exang     oldpeak  
count  261.000000  261.000000  261.000000  261.000000  261.000000  
mean     0.072797    0.113027    0.529871    0.318008    0.122452  
std      0.260302    0.235468    0.219976    0.466597    0.186383  
min      0.000000    0.000000    0.000000    0.000000    0.000000  
25%      0.000000    0.000000    0.370370    0.000000    0.000000  
50%      0.000000    0.000000    0.537037    0.000000    0.000000  
75%      0.000000    0.000000    0.675926    1.000000    0.200000  
max      1.000000    1.000000    1.000000    1.000000    1.000000


The target variable data set (x,y) contains x records across y features:
(261,)

Below is a list of all the variables present in the target variable data set:

num

The predictor variables data set (x,y) contains x records across y features:
(261, 10)

Below is a list of all the variables present in the predictor variable data set:

Index(['age', 'sex', 'cp', 'trestbps', 'chol', 'fbs', 'restecg', 'thalach',
       'exang', 'oldpeak'],
      dtype='object')


### Heart Attacks Data Set Pre-Processing Summary

> <b>Initial</b>
- 294 records by 14 variables
- 14 Numeric Variables and 0 Non-Numeric Variables
- Variety of Ranges

> <b>Cleaned</b>
- 261 records by 11 variables
- 14 Numeric Variables
- Min-Max Standardized

### Post Operation Data Set

In [41]:
# Importing Post Operation data set
Post_Operation = pd.read_csv(pathname + 'Datasets\\Post-Operation.csv', encoding = 'utf-8')

# Checking data set was imported by printing first and last five records
print('\r')
print("Below shows the first five records in the data set:")
display(Post_Operation.head())

# Checking the size of the data set
print('\r')
print("The data set has x records across y variables (x,y):")
print(Post_Operation.shape)

######################### Numeric and Non-Numeric Variables #########################

# Creating a target variable
Post_Operation['Intensive_Care'] = np.where(Post_Operation['ADM-DECS']=="I", 1, 0)

# Dropping variable that directly created target
del Post_Operation['ADM-DECS']

# Describing all numeric variables
print('\r')
print("Below is information about every numeric variable present in the data set:")
display(Post_Operation.describe(include=[np.number]))
print('\r')
print("Notice that "+str(Post_Operation.describe(include=[np.number]).shape[1])+" columns of data contain numeric data.")

# Describing all non-numeric variables
if (Post_Operation.describe(include=[np.number]).shape[1] <  Post_Operation.shape[1]):
    print('\r')
    print("Below is information about every non-numeric variable present in the data set:")
    display(Post_Operation.describe(exclude=[np.number]))
    print('\r')
    print("Notice that "+str(Post_Operation.describe(exclude=[np.number]).shape[1])+" columns of data contain non-numeric data.")
else:
    print('\r')
    print("There are no non-numeric variables within the data set.")
    
######################### Missing Data #########################

# Creating a count of missing values in each column
missing_values = Post_Operation.isnull().sum()

# Displaying the number of missing values in each column that contains a missing value
print('\r')
print("The variables below contain missing values:")
display(missing_values[missing_values>0])

# Dropping all rows with a missing value
Post_Operation = Post_Operation.dropna()

######################### Re-locating and Converting Variables #########################

# Designating Target Variable
Target_Variable = 'Intensive_Care'
Target_Series = Post_Operation[Target_Variable]

# Removing target variable from current location and inserting in collum 0
del Post_Operation[Target_Variable]
Post_Operation.insert(0, Target_Variable, Target_Series)

# Converting target and predictor into numeric variable for learning algorithms later
numeric_encoder = preprocessing.LabelEncoder()
Post_Operation['L-CORE'] = numeric_encoder.fit_transform(Post_Operation['L-CORE'])
Post_Operation['L-SURF'] = numeric_encoder.fit_transform(Post_Operation['L-SURF'])
Post_Operation['L-O2'] = numeric_encoder.fit_transform(Post_Operation['L-O2'])
Post_Operation['L-BP'] = numeric_encoder.fit_transform(Post_Operation['L-BP'])
Post_Operation['SURF-STBL'] = numeric_encoder.fit_transform(Post_Operation['SURF-STBL'])
Post_Operation['CORE-STBL'] = numeric_encoder.fit_transform(Post_Operation['CORE-STBL'])
Post_Operation['BP-STBL'] = numeric_encoder.fit_transform(Post_Operation['BP-STBL'])
Post_Operation['COMFORT'] = numeric_encoder.fit_transform(Post_Operation['COMFORT'])

# Showing the data frame after target variable has been moved to the front
print('\r')
print("Below shows the first five records in the data set after the target variable has been moved and converted:")
display(Post_Operation.head())

######################### Conducting Min-Max Standardization #########################

# Creating a list of numeric variable names
List_of_Numeric_Variables = Post_Operation.describe(include=[np.number]).columns.tolist()

# Creating a for loop to min-max standardize every numeric variable
for variable in List_of_Numeric_Variables:
    Post_Operation[variable] = (Post_Operation[variable]-Post_Operation[variable].min())/(Post_Operation[variable].max()-Post_Operation[variable].min())

# Displaying updated numeric variable statistics to check min max standardization was successful
print('\r')
print("Below is an updated table of numeric variable statistics to demonstrate Min-Max Standardization has been successful:")
display(Post_Operation.describe(include=[np.number]))

######################### Target Variable and Predictor Sets #########################

# Defining Target Variable Column
target_variable_column = Post_Operation[Target_Variable]

# Checking the size of the target variable data set
print('\r')
print("The target variable data set (x,y) contains x records across y features:")
print(target_variable_column.shape)
print('\r')
print("Below is a list of all the variables present in the target variable data set:")
print('\r')
print(target_variable_column.name)

# Creating a data set contains all the predictor variables
predictor_variables_columns = Post_Operation.drop([Target_Variable], axis=1) 

# Checking the size of the target variable data set
print('\r')
print("The predictor variables data set (x,y) contains x records across y features:")
print(predictor_variables_columns.shape)
print('\r')
print("Below is a list of all the variables present in the predictor variable data set:")
print('\r')
print(predictor_variables_columns.columns)

# Storing target variable and predictor variables to arrays for PCA
target_variable_array = target_variable_column.values
predictor_variables_array = predictor_variables_columns.values

######################### Exporting Cleaned Data Set for Analysis #########################

# Exporting cleaned data frame as csv
Post_Operation.to_csv('Post_Operation_Cleaned.csv', index=False)


Below shows the first five records in the data set:


L-CORE L-SURF       L-O2  L-BP SURF-STBL CORE-STBL     BP-STBL COMFORT  \
0    mid    low  excellent   mid    stable    stable      stable      15   
1    mid   high  excellent  high    stable    stable      stable      10   
2   high    low  excellent  high    stable    stable  mod-stable      10   
3    mid    low       good  high    stable  unstable  mod-stable      15   
4    mid    mid  excellent  high    stable    stable      stable      10   

  ADM-DECS  
0        A  
1        S  
2        A  
3       A   
4        A


The data set has x records across y variables (x,y):
(91, 9)

Below is information about every numeric variable present in the data set:


Intensive_Care
count       91.000000
mean         0.032967
std          0.179540
min          0.000000
25%          0.000000
50%          0.000000
75%          0.000000
max          1.000000


Notice that 1 columns of data contain numeric data.

Below is information about every non-numeric variable present in the data set:


L-CORE L-SURF  L-O2 L-BP SURF-STBL CORE-STBL BP-STBL COMFORT
count      91     91    91   91        91        91      91      91
unique      3      3     2    3         2         3       3       5
top       mid    mid  good  mid  unstable    stable  stable      10
freq       58     49    48   58        46        84      47      65


Notice that 8 columns of data contain non-numeric data.

The variables below contain missing values:


Series([], dtype: int64)


Below shows the first five records in the data set after the target variable has been moved and converted:


Intensive_Care  L-CORE  L-SURF  L-O2  L-BP  SURF-STBL  CORE-STBL  BP-STBL  \
0               0       2       1     0     2          0          1        1   
1               0       2       0     0     0          0          1        1   
2               0       0       1     0     0          0          1        0   
3               0       2       1     1     0          0          2        0   
4               0       2       2     0     0          0          1        1   

   COMFORT  
0        1  
1        0  
2        0  
3        1  
4        0


Below is an updated table of numeric variable statistics to demonstrate Min-Max Standardization has been successful:


Intensive_Care     L-CORE     L-SURF       L-O2       L-BP  SURF-STBL  \
count       91.000000  91.000000  91.000000  91.000000  91.000000  91.000000   
mean         0.032967   0.747253   0.675824   0.527473   0.653846   0.505495   
std          0.179540   0.367980   0.389969   0.502011   0.469588   0.502740   
min          0.000000   0.000000   0.000000   0.000000   0.000000   0.000000   
25%          0.000000   0.500000   0.500000   0.000000   0.000000   0.000000   
50%          0.000000   1.000000   1.000000   1.000000   1.000000   1.000000   
75%          0.000000   1.000000   1.000000   1.000000   1.000000   1.000000   
max          1.000000   1.000000   1.000000   1.000000   1.000000   1.000000   

       CORE-STBL    BP-STBL    COMFORT  
count  91.000000  91.000000  91.000000  
mean    0.527473   0.510989   0.115385  
std     0.136680   0.349428   0.236607  
min     0.000000   0.000000   0.000000  
25%     0.500000   0.500000   0.000000  
50%     0.500000   0.500000   0.000000  
75%     0.500000   0.750000   0.250000  
max     1.000000   1.000000   1.000000


The target variable data set (x,y) contains x records across y features:
(91,)

Below is a list of all the variables present in the target variable data set:

Intensive_Care

The predictor variables data set (x,y) contains x records across y features:
(91, 8)

Below is a list of all the variables present in the predictor variable data set:

Index(['L-CORE', 'L-SURF', 'L-O2', 'L-BP', 'SURF-STBL', 'CORE-STBL', 'BP-STBL',
       'COMFORT'],
      dtype='object')


### Post Operation Data Set Pre-Processing Summary

> <b>Initial</b>
- 91 records by 9 variables
- 1 Numeric Variables and 8 Non-Numeric Variable
- Variety of Ranges

> <b>Cleaned</b>
- 91 records by 9 variables
- 9 Numeric Variables
- Min-Max Standardized

### Glass Type Data Set

In [48]:
# Importing Post Operation data set
Glass_Type = pd.read_csv(pathname + 'Datasets\\Glass Type.csv', encoding = 'utf-8')

# Checking data set was imported by printing first and last five records
print('\r')
print("Below shows the first five records in the data set:")
display(Glass_Type.head())

# Checking the size of the data set
print('\r')
print("The data set has x records across y variables (x,y):")
print(Glass_Type.shape)

######################### Numeric and Non-Numeric Variables #########################

# Creating a target variable
Glass_Type['Target_Type'] = np.where(Glass_Type['Type']==6, 1, 0)

# Dropping variable that directly created target
del Glass_Type['Type']

# Describing all numeric variables
print('\r')
print("Below is information about every numeric variable present in the data set:")
display(Glass_Type.describe(include=[np.number]))
print('\r')
print("Notice that "+str(Glass_Type.describe(include=[np.number]).shape[1])+" columns of data contain numeric data.")

# Describing all non-numeric variables
if (Glass_Type.describe(include=[np.number]).shape[1] <  Glass_Type.shape[1]):
    print('\r')
    print("Below is information about every non-numeric variable present in the data set:")
    display(Glass_Type.describe(exclude=[np.number]))
    print('\r')
    print("Notice that "+str(Glass_Type.describe(exclude=[np.number]).shape[1])+" columns of data contain non-numeric data.")
else:
    print('\r')
    print("There are no non-numeric variables within the data set.")
    
######################### Missing Data #########################

# Creating a count of missing values in each column
missing_values = Glass_Type.isnull().sum()

# Displaying the number of missing values in each column that contains a missing value
print('\r')
print("The variables below contain missing values:")
display(missing_values[missing_values>0])

######################### Re-locating and Converting Variables #########################

# Designating Target Variable
Target_Variable = 'Target_Type'
Target_Series = Glass_Type[Target_Variable]

# Removing target variable from current location and inserting in collum 0
del Glass_Type[Target_Variable]
Glass_Type.insert(0, Target_Variable, Target_Series)

# Showing the data frame after target variable has been moved to the front
print('\r')
print("Below shows the first five records in the data set after the target variable has been moved and converted:")
display(Glass_Type.head())

######################### Conducting Min-Max Standardization #########################

# Creating a list of numeric variable names
List_of_Numeric_Variables = Glass_Type.describe(include=[np.number]).columns.tolist()

# Creating a for loop to min-max standardize every numeric variable
for variable in List_of_Numeric_Variables:
    Glass_Type[variable] = (Glass_Type[variable]-Glass_Type[variable].min())/(Glass_Type[variable].max()-Glass_Type[variable].min())

# Displaying updated numeric variable statistics to check min max standardization was successful
print('\r')
print("Below is an updated table of numeric variable statistics to demonstrate Min-Max Standardization has been successful:")
display(Glass_Type.describe(include=[np.number]))

######################### Target Variable and Predictor Sets #########################

# Defining Target Variable Column
target_variable_column = Glass_Type[Target_Variable]

# Checking the size of the target variable data set
print('\r')
print("The target variable data set (x,y) contains x records across y features:")
print(target_variable_column.shape)
print('\r')
print("Below is a list of all the variables present in the target variable data set:")
print('\r')
print(target_variable_column.name)

# Creating a data set contains all the predictor variables
predictor_variables_columns = Glass_Type.drop([Target_Variable], axis=1) 

# Checking the size of the target variable data set
print('\r')
print("The predictor variables data set (x,y) contains x records across y features:")
print(predictor_variables_columns.shape)
print('\r')
print("Below is a list of all the variables present in the predictor variable data set:")
print('\r')
print(predictor_variables_columns.columns)

# Storing target variable and predictor variables to arrays for PCA
target_variable_array = target_variable_column.values
predictor_variables_array = predictor_variables_columns.values

######################### Exporting Cleaned Data Set for Analysis #########################

# Exporting cleaned data frame as csv
Glass_Type.to_csv('Glass_Type_Cleaned.csv', index=False)


Below shows the first five records in the data set:


RI     Na    Mg    Al     Si     K    Ca   Ba   Fe  Type
0  1.52101  13.64  4.49  1.10  71.78  0.06  8.75  0.0  0.0     1
1  1.51761  13.89  3.60  1.36  72.73  0.48  7.83  0.0  0.0     1
2  1.51618  13.53  3.55  1.54  72.99  0.39  7.78  0.0  0.0     1
3  1.51766  13.21  3.69  1.29  72.61  0.57  8.22  0.0  0.0     1
4  1.51742  13.27  3.62  1.24  73.08  0.55  8.07  0.0  0.0     1


The data set has x records across y variables (x,y):
(214, 10)

Below is information about every numeric variable present in the data set:


RI          Na          Mg          Al          Si           K  \
count  214.000000  214.000000  214.000000  214.000000  214.000000  214.000000   
mean     1.518365   13.407850    2.684533    1.444907   72.650935    0.497056   
std      0.003037    0.816604    1.442408    0.499270    0.774546    0.652192   
min      1.511150   10.730000    0.000000    0.290000   69.810000    0.000000   
25%      1.516523   12.907500    2.115000    1.190000   72.280000    0.122500   
50%      1.517680   13.300000    3.480000    1.360000   72.790000    0.555000   
75%      1.519157   13.825000    3.600000    1.630000   73.087500    0.610000   
max      1.533930   17.380000    4.490000    3.500000   75.410000    6.210000   

               Ca          Ba          Fe  Target_Type  
count  214.000000  214.000000  214.000000   214.000000  
mean     8.956963    0.175047    0.057009     0.042056  
std      1.423153    0.497219    0.097439     0.201188  
min      5.430000    0.000000    0.000000     0.000000  
25%      8.240000    0.000000    0.000000     0.000000  
50%      8.600000    0.000000    0.000000     0.000000  
75%      9.172500    0.000000    0.100000     0.000000  
max     16.190000    3.150000    0.510000     1.000000


Notice that 10 columns of data contain numeric data.

There are no non-numeric variables within the data set.

The variables below contain missing values:


Series([], dtype: int64)


Below shows the first five records in the data set after the target variable has been moved and converted:


Target_Type       RI     Na    Mg    Al     Si     K    Ca   Ba   Fe
0            0  1.52101  13.64  4.49  1.10  71.78  0.06  8.75  0.0  0.0
1            0  1.51761  13.89  3.60  1.36  72.73  0.48  7.83  0.0  0.0
2            0  1.51618  13.53  3.55  1.54  72.99  0.39  7.78  0.0  0.0
3            0  1.51766  13.21  3.69  1.29  72.61  0.57  8.22  0.0  0.0
4            0  1.51742  13.27  3.62  1.24  73.08  0.55  8.07  0.0  0.0


Below is an updated table of numeric variable statistics to demonstrate Min-Max Standardization has been successful:


Target_Type          RI          Na          Mg          Al  \
count   214.000000  214.000000  214.000000  214.000000  214.000000   
mean      0.042056    0.316744    0.402684    0.597891    0.359784   
std       0.201188    0.133313    0.122798    0.321249    0.155536   
min       0.000000    0.000000    0.000000    0.000000    0.000000   
25%       0.000000    0.235843    0.327444    0.471047    0.280374   
50%       0.000000    0.286655    0.386466    0.775056    0.333333   
75%       0.000000    0.351514    0.465414    0.801782    0.417445   
max       1.000000    1.000000    1.000000    1.000000    1.000000   

               Si           K          Ca          Ba          Fe  
count  214.000000  214.000000  214.000000  214.000000  214.000000  
mean     0.507310    0.080041    0.327785    0.055570    0.111783  
std      0.138312    0.105023    0.132263    0.157847    0.191056  
min      0.000000    0.000000    0.000000    0.000000    0.000000  
25%      0.441071    0.019726    0.261152    0.000000    0.000000  
50%      0.532143    0.089372    0.294610    0.000000    0.000000  
75%      0.585268    0.098229    0.347816    0.000000    0.196078  
max      1.000000    1.000000    1.000000    1.000000    1.000000


The target variable data set (x,y) contains x records across y features:
(214,)

Below is a list of all the variables present in the target variable data set:

Target_Type

The predictor variables data set (x,y) contains x records across y features:
(214, 9)

Below is a list of all the variables present in the predictor variable data set:

Index(['RI', 'Na', 'Mg', 'Al', 'Si', 'K', 'Ca', 'Ba', 'Fe'], dtype='object')


### Glass Type Data Set Pre-Processing Summary

> <b>Initial</b>
- 214 records by 10 variables
- 10 Numeric Variables and 0 Non-Numeric Variables
- Variety of Ranges

> <b>Cleaned</b>
- 214 records by 10 variables
- 10 Numeric Variables
- Min-Max Standardized